contrib/docker/src/DockerModule.scala

```diff
@@ -113,20 +112,15 @@ trait DockerModule { outer: JavaModule =>
         if (user().isEmpty) "" else s"USER ${user()}"
       ).filter(_.nonEmpty).mkString(sys.props("line.separator"))
 
-      s"""
-         |FROM ${baseImage()}
+      s"""FROM ${baseImage()}
          |$lines
-         |COPY $jarName /$jarName
-         |ENTRYPOINT ["java", "-jar", "/$jarName"]""".stripMargin
+         |COPY ${assembly().path} /${assembly().path.last}
+         |ENTRYPOINT ["java", "-jar", "/${assembly().path.last}"]""".stripMargin
     }
 
     final def build = T {
-      val dest = T.dest
-
-      val asmPath = outer.assembly().path
-      os.copy(asmPath, dest / asmPath.last)
-
-      os.write(dest / "Dockerfile", dockerfile())
+      val dockerfilePath = T.dest / "Dockerfile"
+      os.write(dockerfilePath, dockerfile().replace(os.pwd.toString, ""))
```

Comment:
We have to scale down all mentions of absolute paths to the project root.

Reply:
`T.workspace` is a special accessor to the project root directory.

```suggestion
      os.write(dockerfilePath, dockerfile().replace(T.workspace.toString, ""))
```